modules/programs/fish.nix

```diff
@@ -12,6 +12,23 @@ let
     mapAttrsFlatten (k: v: "alias ${k} '${v}'") cfg.shellAliases
   );
 
+  babelfish = pkgs.callPackage ../../pkgs/shells/fish/plugins/babelfish.nix { };
+
+  envShellInit = pkgs.writeText "shellInit" cfge.shellInit;
+
+  envLoginShellInit = pkgs.writeText "loginShellInit" cfge.loginShellInit;
+
+  envInteractiveShellInit = pkgs.writeText "interactiveShellInit" cfge.interactiveShellInit;
+
+  sourceEnv = useTranslated: path:
+  if useTranslated then
+    "source ${path}"
+  else
+    ''
+      set fish_function_path ${pkgs.fishPlugins.foreign-env}/share/fish-foreign-env/functions $fish_function_path
```

Comment:
Similar to the babelfish package, this should still handle the case where fishPlugins isn't available in nixpkgs yet. That's what is causing the tests to fail. eg.

```nix
let
  fenv = pkgs.fish-foreign-env or pkgs.fishPlugins.foreign-env;
in
  # ...
```

Ideally an alias would be added in nixpkgs however, just renaming stuff isn't really something projects outside of the repo should have deal with without some compatibility window.

Reply:
That's a very clean way to do it. I'll incorporate that into this branch, along with the other changes (like selecting which method to use—fenv or babelfish—) from the nixOS module and push up a new version.